File: CafeEase/CafeEase.Model/Cafe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CafeEase.Model
{
    public class Cafe
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string CityName { get; set; } = string.Empty;
        public int CityId { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string? PhoneNumber { get; set; }
        public string? WorkingHours { get; set; }
        public bool IsActive { get; set; }
        public double? DistanceKm { get; set; }
    }
}

File: CafeEase/UI/cafeease_desktop/lib/models/cafe.dart
import 'package:json_annotation/json_annotation.dart';

part 'cafe.g.dart';

@JsonSerializable()
class Cafe {
  final int? id;
  final String? name;
  final String? address;
  final String? cityName;
  final int? cityId;
  final double? latitude;
  final double? longitude;
  final String? phoneNumber;
  final String? workingHours;
  final bool? isActive;
  final double? distanceKm;

  Cafe({
    this.id,
    this.name,
    this.address,
    this.cityName,
    this.cityId,
    this.latitude,
    this.longitude,
    this.phoneNumber,
    this.workingHours,
    this.isActive,
    this.distanceKm,
  });

  factory Cafe.fromJson(Map<String, dynamic> json) => _$CafeFromJson(json);
  Map<String, dynamic> toJson() => _$CafeToJson(this);
}
